File: quick_sort/quick_sort.js
function quick_sort(array) {
  if (array.length < 2) {
    return array;
  } else {
    const pivot = array[0];
    const left = [];
    const right = [];
    for (let i = 1; i < array.length; i++) {
      if (array[i] < pivot) {
        left.push(array[i]);
      } else {
        right.push(array[i]);
      }
    }
    return quick_sort(left).concat([pivot], quick_sort(right));
  }
}

File: quick_sort/quick_sort.py
def quick_sort(array):
    if len(array) < 2:
        return array
    else:
        pivot = array[0]

        less = [x for x in array[1:] if x <= pivot]

        greater = [x for x in array[1:] if x > pivot]

        return quick_sort(less) + [pivot] + quick_sort(greater)
    
print(quick_sort([10, 5, 2, 3]))

